crates/egui/src/widgets/progress_bar.rs

```diff
@@ -95,16 +112,15 @@ impl Widget for ProgressBar {
             }
 
             let visuals = ui.style().visuals.clone();
-            let rounding = outer_rect.height() / 2.0;
+            let is_custom_rounding = rounding.is_some();
+            let corner_radius = outer_rect.height() / 2.0;
+            let rounding = rounding.unwrap_or_else(|| corner_radius.into());
             ui.painter()
                 .rect(outer_rect, rounding, visuals.extreme_bg_color, Stroke::NONE);
-            let inner_rect = Rect::from_min_size(
-                outer_rect.min,
-                vec2(
-                    (outer_rect.width() * progress).at_least(outer_rect.height()),
-                    outer_rect.height(),
-                ),
-            );
+            let min_width = 2.0 * rounding.sw.at_least(rounding.nw).at_most(corner_radius);
```

Comment:
why only look at `sw` and `nw` here? Wouldn't it make sense to use something like `rounding.max()` instead?

Reply:
Since this `min_width` is meant to make the rounding look right when the progress bar is empty, I figured only the left side mattered. But maybe it would be better to just use `max()`. I'll do a quick test and commit it.

Reply:
So there is no method `Rounding::max` currently. Should I add one?

It doesn't seem to be necessary, though. The problem this min_width avoids is that the filled rect is flattened when it is too small, and then it ends up appearing outside the background rect, like this:

![image](https://github.com/emilk/egui/assets/43298013/00f53f23-746d-48c6-bf7e-6743be4fc863)

Actually, if there was a way to avoid this overflow without setting a minimum width, I think a lot of people would prefer that (see for example [#3999](https://github.com/emilk/egui/discussions/3399)). But I don't know if it's possible.